src/client/components/Header.tsx

```diff
@@ -16,17 +16,39 @@ const useStyles = makeStyles((theme: Theme) =>
   }),
 );
 
+const onTextFieldUpdate = (toUpdate : React.Dispatch<React.SetStateAction<string>>) => {
+  return (e : React.ChangeEvent<HTMLInputElement>) => {
+      toUpdate(e.target.value);
+  };
+}
+
+
+
 export const Header: React.FunctionComponent = (props) => {
   const {data, loading} = useMeQuery();
   const [logout, {client}] = useLogoutMutation();
+  const [changePassword] = useChangePasswordMutation();
   const isLoggedIn = data && data.me;
+  const [open, setOpen] = useState(false);
+  const [oldPassword, setOldPassword] = useState("");
+  const [newPassword, setNewPassword] = useState("");
+  const [confirmPassword, setConfirmPassword] = useState("");
+  
 
   const logOutUser = async () => {
     await logout();
     setAccessToken("");
     await client.resetStore();
   }
 
+  const handleConfirmPassword = async (oldPassword, newPassword, confirmPassword) => {
+    if (newPassword === confirmPassword){
+      changePassword({variables: {oldPassword, newPassword}})
```

Comment:
`return await changePassword` because this is an asynchronous function because this function should return whether or not the changePassword was successful. We will someday have to handle these return values to show whether or not it was successful or not

Reply:
This still